modules/scheduler/gke-cluster/variables.tf

```diff
@@ -333,3 +333,9 @@ variable "additional_networks" {
     }))
   }))
 }
+
+variable "max_pods_per_node" {
```

Comment:
Does `max_pods_per_node` in node-pool overrides cluster's `default_max_pods_per_node`? if yes, I would use `default_max_pods_per_node` as name for cluster.